### ISSUE TITLE
Move PlayMode tests running the main stories into separate GitHub Actions job

### ISSUE THREAD

**Issue by ViMaSter** [Studio-Lovelies/GG-JointJustice-Unity#247]
## Description
Right now, the `PlayMode Scene` tests include a full playthrough of all (currently available) main story cases of the game. 

Naturally, these take some time; around 5-6 minutes for case 1 alone, resulting in a significantly longer job duration over the `PlayMode Scripts` or `EditMode`.
<sub>[`Scripts` and `Scenes` labels are swapped; tracked under #246](https://github.com/Studio-Lovelies/GG-JointJustice-Unity/issues/246)</sub>
![image](https://user-images.githubusercontent.com/1689033/161402176-9c96f830-13f0-4722-954e-e39d649cec99.png)

To alleviate this, these "case playthrough" tests should ideally be moved into a separate folder on the same level as `Scripts` and `Scenes` and split off into another job on the GitHub Actions pipeline, so they can run in parallel.
For now, the WebGL build already takes around 25 minutes, so this is does not yet have an impact on build times, but will as soon as we add more cases.

## Acceptance Criteria
This story will be finished when the files located at `/unity-ggjj/Assets/Tests/PlayModeTests/Scenes/NarrativeScripts` are moved to `/unity-ggjj/Assets/Tests/PlayModeTests/Cases` and GitHub Runners execute these tests in parallel to the `PlayMode Scripts` and `PlayMode Scene` runs.

## "Nice to have" sub-items
This task has two potential "nice to have" subtasks. These **can** be achieve in one go, but **don't have to**.  
⚠When tackling only parts of these issues, please create new issues for the not-yet-implemented "nice to have" items for visibility. ⚠

### Running each case in parallel
As we add more cases, these times will increase even more. While not necessary yet, the solution could already account for this, by allowing **multiple cases to run in parallel, rather than sequentially**. This could be achieved [using GitHub action matrices](https://docs.github.com/en/actions/using-jobs/using-a-build-matrix-for-your-jobs), like we already do [for our various build platform targets](https://github.com/Studio-Lovelies/GG-JointJustice-Unity/blob/develop/.github/workflows/build.yml#L231-L245).

### Automatic discovery of subfolders
Right now, parallelization of tests is achieved, by cloning the repo on the runner and [deleting the irrelevant tests by hand](https://github.com/Studio-Lovelies/GG-JointJustice-Unity/blob/develop/.github/workflows/build.yml#L143-L160).  
It should be possible to achievable this automatically:
1. Find each subfolder inside `/unity-ggjj/Assets/Tests/PlayModeTests/*`
2. Generate a list of all folders
3. Storing it as an output variable in one job
4. Use this variable to populate a matrix (as mentioned in [Running each case in parallel](Running each case in parallel))
5. For each entry in that matrix
  1. download the repo
  2. delete all **except** that folder
  3. run the tests
  4. store the test results and coverage reports
